tests/sources/test_poetry_pyproject.py

```diff
@@ -0,0 +1,289 @@
+    fp = StringIO(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+    """))
+    fp.name = "pyproject.toml"
+    result = update_python_requires(fp, v(['3.5', '3.6', '3.7']))
+    assert result is None
+    assert capsys.readouterr().err == ""
+
+
+def test_update_python_requires_preserves_style(fix_max_python_3_version):
+    fix_max_python_3_version(2)
+    fp = StringIO(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ">=2.7,!=3.0.*"
+    """))
+    fp.name = "pyproject.toml"
+    result = update_python_requires(fp, v(['2.7', '3.2']))
+    assert "".join(result) == textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ">=2.7,!=3.0.*,!=3.1.*"
+    """)
+
+
+def test_update_python_requires_multiline(fix_max_python_3_version):
+    fix_max_python_3_version(2)
+    fp = StringIO(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ', '.join([
+                '>=2.7',
+                '!=3.0.*',
+            ])
+    """))
+    fp.name = "pyproject.toml"
+    result = update_python_requires(fp, v(['2.7', '3.2']))
+    assert result is not None
+    assert "".join(result) == textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ', '.join([
+                '>=2.7',
+                '!=3.0.*',
+                '!=3.1.*',
+            ])
+    """)
+
+
+def test_update_python_requires_multiline_variations(fix_max_python_3_version):
+    fix_max_python_3_version(2)
+    fp = StringIO(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ",".join([
+                ">=2.7",
+                "!=3.0.*",
+            ])
+    """))
+    fp.name = "pyproject.toml"
+    result = update_python_requires(fp, v(['2.7', '3.2']))
+    assert result is not None
+    assert "".join(result) == textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ",".join([
+                ">=2.7",
+                "!=3.0.*",
+                "!=3.1.*",
+            ])
+    """)
+
+
+def test_update_python_requires_multiline_error(capsys):
+    fp = StringIO(textwrap.dedent("""\
+        [tool.poetry]
+            name='foo',
+            [tool.poetry.dependencies]
+                python = ', '.join([
+                '>=2.7',
+                '!=3.0.*'])
+    """))
+    fp.name = "pyproject.toml"
+    result = update_python_requires(fp, v(['2.7', '3.2']))
+    assert result == fp.getvalue().splitlines(True)
+    assert (
+        "Did not understand python_requires formatting in python dependency"
+        in capsys.readouterr().err
+    )
```

Comment:
I would like to see a unit test that modifying the toml file leaves comments in place.

Reply:
added dedicated test